demos/python_demos/object_detection_demo_faceboxes/detector.py

```diff
@@ -0,0 +1,200 @@
+            else:
+                list_x = [f_x + 0.5]
+                list_y = [f_y + 0.5]
+            anchors.extend(self.calculate_anchors(list_x, list_y, min_size, image_size, step))
+        return anchors
+
+    def prior_boxes(self, feature_maps, image_size):
+        anchors = []
+        for k, f in enumerate(feature_maps):
+            for i, j in itertools.product(range(f[0]), range(f[1])):
+                if k == 0:
+                    anchors.extend(self.calculate_anchors_zero_level(j, i, self.min_sizes[k],
+                                                                     image_size, self.steps[k]))
+                else:
+                    anchors.extend(self.calculate_anchors([j + 0.5], [i + 0.5], self.min_sizes[k][0],
+                                                          image_size, self.steps[k]))
+        anchors = np.clip(anchors, 0, 1)
+
+        return anchors
+
+    @staticmethod
+    def nms(x1, y1, x2, y2, scores, thresh, include_boundaries=True, keep_top_k=None):
+        b = 1 if include_boundaries else 0
+
+        areas = (x2 - x1 + b) * (y2 - y1 + b)
+        order = scores.argsort()[::-1]
+
+        if keep_top_k:
+            order = order[:keep_top_k]
+
+        keep = []
+        while order.size > 0:
+            i = order[0]
+            keep.append(i)
+
+            xx1 = np.maximum(x1[i], x1[order[1:]])
+            yy1 = np.maximum(y1[i], y1[order[1:]])
+            xx2 = np.minimum(x2[i], x2[order[1:]])
+            yy2 = np.minimum(y2[i], y2[order[1:]])
+
+            w = np.maximum(0.0, xx2 - xx1 + b)
+            h = np.maximum(0.0, yy2 - yy1 + b)
+            intersection = w * h
+
+            union = (areas[i] + areas[order[1:]] - intersection)
+            overlap = np.divide(intersection, union, out=np.zeros_like(intersection, dtype=float), where=union != 0)
+
+            order = order[np.where(overlap <= thresh)[0] + 1]  # pylint: disable=W0143
+
+        return keep
+
+    @staticmethod
+    def resize_boxes(detections, image_size):
+        h, w = image_size
+        detections['x_mins'] = [x_min * w for x_min in detections['x_mins']]
+        detections['x_maxs'] = [x_max * w for x_max in detections['x_maxs']]
+        detections['y_mins'] = [y_min * h for y_min in detections['y_mins']]
+        detections['y_maxs'] = [y_max * h for y_max in detections['y_maxs']]
+        return detections
+
+    def preprocess(self, image):
+        return cv2.resize(image, (self.input_width, self.input_height))
+
+    def infer(self, image):
+        t0 = cv2.getTickCount()
+        inputs = {self._input_layer_name: image}
+        output = self._exec_model.infer(inputs=inputs)
+        self.infer_time = (cv2.getTickCount() - t0) / cv2.getTickFrequency()
+        return output
+
+    def postprocess(self, raw_output, image_sizes):
+        boxes, scores = raw_output
+
+        detections = {'labels': [], 'scores': [], 'x_mins': [], 'y_mins': [], 'x_maxs': [], 'y_maxs': []}
```

Comment:
Consider using namedtuple instead. Does it make more sense to keep a list of detected objects instead of lists containing a single type of property?

Reply:
Replaced dict with namedtuple

Reply:
So is there a reason to keep lists containing a single type of property instead of list of detected objects in `detections`?
If you keep list of detected objects you don't need that `for label, score, x_min, y_min, x_max, y_max in zip(*detections):` it will be just `for label, score, x_min, y_min, x_max, y_max in detections:`

Reply:
Done